ui/src/message_popup/author/new_question_page.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+import QuestionPage from './question_page.jsx';
+import {allArchivedQuestions} from './archived_questions.js';
+import {allQuestions} from '../questions.js';
+import * as Api from '../../helpers/api.js';
+
+
+export default React.createClass({
+  displayName: 'NewQuestionPage',
+
+  getInitialState(){
+    return ({
+      loaded: false,
+      allDatabaseQuestions: {
+        currentQuestions: allQuestions,
+        archivedQuestions: allArchivedQuestions
+      }
+    });
+  },
+
+  componentDidMount(){
+    Api.questionsQuery().end(this.onQuestionsReceived);
+  },
+
+  onQuestionsReceived(err, response){
+    const questions = JSON.parse(response.text).questions;
+    if(questions !== undefined){
```

Comment:
Yes, I initially tried checking for `err` but I found that it kept being null even the database never connected. Since I'm not too familiar with how it should have worked, I decided to do a workaround. 

With this, it passes down `allDatabaseQuestions` as a prop. Since the `onQuestionsReceived` only updates the `allDatabaseQuestions` prop, if it is initially set as something else, setting loaded to true should not render anything undefined but what was set in `getInitialState` until overridden by a successful database query.